module-system/sov-modules-api/src/lib.rs

```diff
@@ -146,6 +146,7 @@ pub trait Spec {
         + PublicKey
         + Serialize
         + for<'a> Deserialize<'a>
+        + ::schemars::JsonSchema
```

Comment:
Two related questions here :) 
1. Do we need this bound to satisfy the compiler? (Often you can get away without these bounds since you'll always fill in a concrete `Context` implementation before trying to serialize schemas)
2. If not, is there ever a situation where a rollup wouldn't want to support JsonSchema?

Reply:
Good question(s). I'd imagine there are situations where a user doesn't care about generating JSON Schema; maybe when authoring a new module it's annoying to have to deal with this. We not enforce modules to implement `JsonSchema`, but `PublicKey` and `Signature` must implement it. It's not ideal.

We need these bounds to satisfy the compiler because `ModuleCallJson` is implemented for all `C: Context`. I think we could add a macro attribute to specify the type bounds, much like `schemars` already does, and this way we wouldn't have to change any type bounds inside `Spec`. Something of this sort:

```rust
[cfg_attr(
    feature = "native",
    derive(sov_modules_macros::ModuleCallJsonSchema),
    module_call_json_schema(bounds = "C::Signature: schemars::JsonSchema")
)]
#[derive(ModuleInfo, Clone)]
pub struct Election<C: sov_modules_api::Context> {
    #[address]
    pub address: C::Address,
    // ...
}
```

As an alternative, I can try and generate type bounds on `Signature` and `PublicKey` by default and only have the macro attribute to opt-out if the module authors wants to. That seems like the best solution to me.

Reply:
Tracked in https://github.com/Sovereign-Labs/sovereign-sdk/issues/539.